specifyweb/frontend/js_src/lib/components/SchemaConfig/UniquenessRules.tsx

```diff
@@ -0,0 +1,437 @@
+    );
+  }, [tableRules]);
+
+  const handleRuleValidation = React.useCallback(
+    (newRule: UniquenessRule, index: number) => {
+      const filteredRule: UniquenessRule = {
+        ...newRule,
+        fields: newRule.fields.filter(
+          (field, index) =>
+            newRule.fields.map(({ name }) => name).indexOf(field.name) === index
+        ),
+        scopes: newRule.scopes.filter(
+          (scope, index) =>
+            newRule.scopes.map(({ name }) => name).indexOf(scope.name) === index
+        ),
+      };
+      loading(
+        validateUniqueness(
+          container.name as keyof Tables,
+          filteredRule.fields
+            .filter((field) => field.name !== '')
+            .map((field) => field.name) as unknown as RA<never>,
+          filteredRule.scopes
+            .filter(({ name }) => name !== databaseFieldName)
+            .map(({ name }) => name) as unknown as RA<never>
+        ).then((data) => {
+          const isNewRule = index > tableRules.length;
+          setTableRules((previous) =>
+            isNewRule
+              ? [...previous!, { rule: filteredRule, duplicates: data }]
+              : replaceItem(tableRules, index, {
+                  rule: filteredRule,
+                  duplicates: data,
+                })
+          );
+
+          return filteredRule;
+        })
+      );
+    },
+    [container.name, loading, tableRules, setTableRules]
+  );
+
+  return (
+    <Dialog
+      buttons={
+        <>
+          <Button.DialogClose>{commonText.close()}</Button.DialogClose>
+          {saveBlocked ? (
+            <Button.Danger
+              className="cursor-not-allowed"
+              onClick={(): void => undefined}
+            >
+              {icons.exclamation}
+              {commonText.save()}
+            </Button.Danger>
+          ) : (
+            <Submit.Save
+              onClick={(): void => {
+                loading(
+                  ajax(
+                    `/businessrules/uniqueness_rules/${schema.domainLevelIds.discipline}/`,
+                    {
+                      // eslint-disable-next-line @typescript-eslint/naming-convention
+                      headers: { Accept: 'application/json' },
+                      method: tableRules.some(({ rule }) => rule.id === null)
+                        ? 'POST'
+                        : 'PUT',
+                      body: {
+                        rules: tableRules.map(({ rule }) => rule),
+                        removed:
+                          getUniquenessRules()!
+                            [strictGetModel(container.name).name]?.filter(
+                              ({ rule: oldRule }) =>
+                                !tableRules
+                                  .map(({ rule }) => rule.id)
+                                  .includes(oldRule.id)
+                            )
+                            .map(({ rule }) => rule) ?? [],
```

Comment:
Would it make sense to do this on the backend? Look at how backend handles to-many dependent resources deletion:

https://github.com/specify/specify7/blob/685432d865436dc8dda8f51005cb2a400d8e38e4/specifyweb/specify/api.py#L560-L580

So, it will be something like checking which uniqueness rules no longer exist in the current request.

Or will that further complicate things?

Reply:
Yes, checking and removing unused rules should probably be done on the backend. 
It shouldn't complicate things much further.

Reply:
Resolved in https://github.com/specify/specify7/pull/4215/commits/fb7546ccc9d9ce6e1268465ee3802be045c69cd1. 

The only concern I have with this approach is for those who use the API. 
To delete a rule now, they have to send a request which contains every rule save for the one they want to delete. 
This may provide unintentional behavior of removing rules when they just wished to update some other rules. 
Of course we can provide documentation on the endpoint about this behavior, but it still seems like a lot of work and additional understanding would be needed for something that should be at least slightly more trivial. 

Is this a valid concern?

Reply:
It makes it very microscopically harder. And also super simple to implement by anyone, if they are technical enough to already be using the endpoints.